src/postprocess/band_structure.jl

```diff
@@ -99,35 +83,37 @@ end
 end
 
 
-function split_into_branches(kcoords, data::Dict, klabels::Dict)
+function kdistances_and_ticks(kcoords, klabels::Dict)
```

Comment:
Indeed, I hadn't thought of the problem. Thanks for pointing it out.

If the band discontibuity is the only problem, what about storing only the indices of the co tinuous k points as a vector of ranges, e.g. [1:5, 6:8], and using them for plotting?

Reply:
That would probably be a simple and sufficient solution, I agree.

Reply:
On that note: I also find the `klabels` dict a bit clunky. How about changing it to a list of pairs, similar what Brillouin's `kp.points` contains. What to use there depends of course what make the overall implementation and handling of the data generation and plotting easiest.

Reply:
I re-introduced branches as a list of ranges, so the points are not connected at discontinuous k points (see the figure below).

> On that note: I also find the klabels dict a bit clunky. How about changing it to a list of pairs, similar what Brillouin's kp.points contains. What to use there depends of course what make the overall implementation and handling of the data generation and plotting easiest.

I do not understand this point, could you clarify? `klabels` is a mapping from k point labels to the vector. Since this mapping is not ordered, I think a dict is more adequate than a list of pairs. Also, Brillouin's `kp.points` is also a dict (only difference is that the keys are symbols, not strings): https://github.com/thchr/Brillouin.jl/blob/master/src/KPaths.jl#L55-L61

Reply:
Ah sorry, then that's an error on my side. In that case I am all for keeping the Dict (with the additional suggestion to not convert the keys to string).